src/components/RoomPage/EditRoomTable.jsx

```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect } from 'react';
+import EditIcon from './EditIcon'; 
+import BinIcon from './BinIcon';
+import axios from "../../api/AxiosNoqApi.js";
+import useLogin from "../../hooks/useLogin";
+
+const EditRoomTable = () => {
+  const { login } = useLogin();
+  const [id, setId] = useState(0);
+  const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
+  const [host_id, setHostId] = useState('');
+  const [beds, setBeds] = useState('');
+  const [type, setType] = useState('');
+  const [requirements, setRequirements] = useState('');
+  const [rows, setRows] = useState([]);
+  const [editIndex, setEditIndex] = useState(null);
+
+
+  useEffect(() => {
+    setHostId(login.host_id);
+    fetchRooms();
+  }, []);
+
+  const fetchRooms = async () => {
+    try {
+      const response = await axios.get('api/host/products');
```

Comment:
Should use "api/host/hosts/{host_id}/products" as this API call returns all products for all hosts. Another option is to change the backend API "api/host/products" to return only products for the specific host and remove "api/host/hosts/{host_id}/products" from the backend. Either fix is ok for me.